### PR TITLE
rpc: Add debugLevel support in wallet (#1184)

### PR DESCRIPTION
**Pull request by ggoranov**
Previously, when debuglevel request is sent dcrwallet tried to
perform RPC passthrough to dcrd. This commit allows dcrwallet
to set debug level per subsystem.

### CODE REVIEW

File: dcrwallet.go [hunk 1]
```diff
@@ -287,6 +287,18 @@ func run(ctx context.Context) error {
 				requestShutdown()
 			}
 		}()
+		// As the server is another package an chan is used to access
+		// functions from main
+		go func() {
+			for debugLevel := range jsonRPCServer.RequestProcessDebugLevel() {
+				if err := parseAndSetDebugLevels(debugLevel); err != nil {
+					err := errors.Errorf("%v", err.Error())
+					fmt.Fprintln(os.Stderr, err)
```
Comment (jrick) [inline dcrwallet.go]: I'd like to see this error returned to the RPC caller instead of just printed to output.

**Comment by ggoranov** [comment]
Thanks for the suggestion. Ready for review.

File: rpc/legacyrpc/server.go [hunk 2]
```diff
@@ -598,3 +604,22 @@ func (s *Server) requestProcessShutdown() {
...
+// ResponseProcessDebugLevel returns response chan that can store an empty string
+// if no error is found or error description string
+func (s *Server) ResponseProcessDebugLevel() chan<- string {
+	return s.responseDebugLevelChan
+}
+
+// setRequestProcessDebugLevel sends debug level to request chan and waits for
+// response message
+func (s *Server) setRequestProcessDebugLevel(debugLevel string) string {
+	s.requestDebugLevelChan <- debugLevel
```
Comment (jrick) [inline rpc/legacyrpc/server.go]: pass an error channel here too. from main, send a non-nil error if the debug level spec was invalid.  you can either send a nil error, or close the channel, if the log level is successfully changed.  By doing this we prevent a logic race where two callers both calling debuglevel at the same time might not receive the correct error.

Comment (ggoranov) [reply rpc/legacyrpc/server.go]: I thought I had considered the logic race situation but seems I've not handled it fully.  Another commit is ready for review.

**Comment by ggoranov** [comment]
@jrick Do you have more ideas for improvements with the latest commit? No rush, just asking.

**Pull request closed without merging**